File: JournalLabs.API/BLL/KindOfWorkService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using JournalLabs.API.DAL.Repositories;
using JournalLabs.API.Models;

namespace JournalLabs.API.BLL
{
    public class KindOfWorkService
    {
        private KindOfWorkRepository _kindOfWorkRepository;

        public KindOfWorkService()
        {
            _kindOfWorkRepository = new KindOfWorkRepository();
        }

        public void CreateKindOfWork(KindOfWork kindOfWorkModel)
        {
            kindOfWorkModel.Id = Guid.NewGuid();
            _kindOfWorkRepository.CreateKindOfWork(kindOfWorkModel);
        }

        public void UpdateKindOfWork(KindOfWork kindOfWorkModel)
        {
            _kindOfWorkRepository.UpdateKindOfWork(kindOfWorkModel);
        }

        public KindOfWork GetKindOfWorkById(string kindOfWorkId)
        {
            return _kindOfWorkRepository.GetKindOfWorkById(kindOfWorkId);
        }
        public bool DeleteKindOfWorkById(string id)
        {
            return _kindOfWorkRepository.DeleteKindOfWorkById(id);
        }
        public bool UpdateVisibleKindOfWork(string idKindOfWork, bool isKindOfWorkVisible)
        {
            return _kindOfWorkRepository.UpdateVisibleKindOfWork(idKindOfWork, isKindOfWorkVisible);
        }
        public bool UpdateVisibleKindOfWorkForStudent(string idKindOfWork, bool isKindOfWorkVisibleForStudent)
        {
            return _kindOfWorkRepository.UpdateVisibleKindOfWorkForStudent(idKindOfWork, isKindOfWorkVisibleForStudent);
        }
        
    }
}

File: JournalLabs.Web/ClientApp/app/shared/kind-of-work.service.ts
import { Injectable, Inject } from '@angular/core';
import { Http, URLSearchParams } from '@angular/http';
import { APP_BASE_HREF } from '@angular/common';
import { ORIGIN_URL } from './constants/baseurl.constants';
import { KindOfWork } from '../models/kind-of-work';
import { TransferHttp } from '../../modules/transfer-http/transfer-http';
import { Observable } from 'rxjs/Observable';
import { REQUEST } from './constants/request';

@Injectable()
export class KindOfWorkService {
  constructor(
    private transferHttp: TransferHttp, // Use only for GETS that you want re-used between Server render -> Client render
    private http: Http, // Use for everything else
    @Inject(REQUEST) private baseUrl: string) {

  }

  getKindOfWorks(): Observable<any> {
    // ** TransferHttp example / concept **
    //    - Here we make an Http call on the server, save the result on the window object and pass it down with the SSR,
    //      The Client then re-uses this Http result instead of hitting the server again!

    //  NOTE : transferHttp also automatically does .map(res => res.json()) for you, so no need for these calls
    return this.transferHttp.get(`${this.baseUrl}/api/KindOfWork/GetKindOfWorks`);
  }

  getKindOfWork(id: string): Observable<any> {
    return this.transferHttp.get(`${this.baseUrl}/api/KindOfWork/GetKindOfWorkById` + id);
  }

  deleteKindOfWork(id: string): Observable<any> {
    return this.transferHttp.get(`${this.baseUrl}/api/KindOfWork/DeleteKindOfWorkById?Id=` + id);
  }

  updateKindOfWork(kindOfWork: KindOfWork): Observable<any> {
    return this.http.post(`${this.baseUrl}/api/KindOfWork/UpdateKindOfWork`, kindOfWork);
  }

  addKindOfWork(kindOfWork: KindOfWork): Observable<any> {
    return this.http.post(`${this.baseUrl}/api/KindOfWork/CreateKindOfWork`, kindOfWork)
  }

  updateVisibleKindOfWork(idKindOfWork: string, isVisible: boolean): Observable<any> {
    return this.transferHttp.get(`${this.baseUrl}/api/KindOfWork/UpdateVisibleKindOfWork?idKindOfWork=${idKindOfWork}&isKindOfWorkVisible=${isVisible}`);
  }
  updateVisibleKindOfWorkForStudent(idKindOfWork: string, isVisible: boolean): Observable<any> {
    return this.transferHttp.get(`${this.baseUrl}/api/KindOfWork/UpdateVisibleKindOfWorkForStudent?idKindOfWork=${idKindOfWork}&isKindOfWorkVisibleForStudent=${isVisible}`);
  }
}
